unittests/allplatformstests.py

```diff
@@ -1358,12 +1358,22 @@ def test_lto_mode(self):
         self.build()
         self.run_tests()
 
-        expected = set(cc.get_lto_compile_args(threads=8, mode='thin'))
+        expected_compile = set(cc.get_lto_compile_args(threads=8, mode='thin'))
+        expected_link = set(cc.get_lto_link_args(threads=8, mode='thin'))
+        # Some platforms don't actually need any lto-specific link options. I.e. for lld-link -
+        # "When using lld-link, the -flto option need only be added to the compile step"
+        # In which case, duplicating the compile lto options for the linker args just produces
+        # unnecessary warnings, so we expect the compile options (but not necessarily the link
+        # options) to be non-empty.
+        self.assertTrue(len(expected_compile) > 0, f'No LTO compile args added "{cc.id}" compiler')
```

Comment:
A set with length 0 is always false, a set with length >0 is always true... I think the usual style for unittest asserts would be:
```python
        self.assertTrue(expected_compile, f'No LTO compile args added "{cc.id}" compiler')
```

The practical difference here is in the logging. When you use len() you get this error:
```
AssertionError: False is not true : No LTO compile args added "{cc.id}" compiler
```

When directly asserting `self.assertTrue(expected_compile)` you get this test failure message:

```
AssertionError: set() is not true: No LTO compile args added "{cc.id}" compiler
```

Although strictly speaking the only crucial bit is the explicit message passed, it doesn't hurt to permit the internal diagnostic pretty-print the actual value passed to the assertion method.

Also, it's shorter to type. :)

Reply:
Thanks. Done.